### ISSUE TITLE
self.non_trainable_weights not working correctly

### ISSUE THREAD

**Issue by AdityaGudimella** [fchollet/keras#3063]
As far as I understand, when I write a layer, and add some weights to self.non_trainable_weights, it should add these weights to the gradient function as T.grad(loss, wrt, consider_constant=non_trainable_weights). Keras is not doing that for my layer, and I don't understand why. Please help me out.
I'm adding the non_trainable_weights in the call function of the layer

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by ChristianThomae** [comment]
Try adding them in the build method, thats where it is typically done.
Use **non**_trainable_weights just for parameters which **aren't** intended to be trained by the optimizer.
So trainable_weights might fit better for your intention of adding them to the gradient.

**Comment by AdityaGudimella** [comment]
I don't want to add them to the gradient. I want to add a few variables to the consider_constant argument of theano grad method. I was able to do this by modifying the keras source code, but was wondering if it should be the default way non_trainable_weights work.

**Comment by ChristianThomae** [comment]
I'm sorry, i misunderstood you there.
Can't provide you an answer here, but that is a good question.
